library/func_misc.py

```diff
@@ -153,3 +153,55 @@ def Call(self, pos_args, named_args):
 
         raise error.TypeErr(x, 'len() expected Str, List, or Dict',
                             loc.Missing)
+
+
+class Join(vm._Callable):
+
+    def __init__(self):
+        # type: () -> None
+        pass
+
+    def Call(self, pos_args, named_args):
+        # type: (List[value_t], Dict[str, value_t]) -> value_t
+
+        r = typed_args.Reader(pos_args, named_args)
+        li = r.PosList()
+
+        delim = r.NamedStr('delim', '')
+        if len(pos_args): # reader has a reference
+            delim = r.PosStr()
+
+        r.Done()
+
+        strs = []  # type: List[str]
+        for i, el in enumerate(li):
+            msg = 'join() items must be Str but item %d is %s' % (
+                i, value_str(el.tag()))
+            strs.append(val_ops.ToStr(el, msg, loc.Missing))
+
+        return value.Str(delim.join(strs))
+
+
+class Maybe(vm._Callable):
+
+    def __init__(self):
+        # type: () -> None
+        pass
+
+    def Call(self, pos_args, named_args):
+        # type: (List[value_t], Dict[str, value_t]) -> value_t
+
+        r = typed_args.Reader(pos_args, named_args)
+        val = r.PosValue()
+        r.Done()
+
+        if val == value.Undef or val == value.Null:
+            return value.List([])
+
+        s = val_ops.ToStr(
```

Comment:
Hmm looks like the original didn't convert to string, but it may not have been well thought out

Hm let me make sure there is a placeholder for documenting all this stuff

Reply:
Actually the intended usage is `@[maybe(x)]`, and `@[]` already does the string conversion if necessary -- it calls `val_ops.Stringify()`, so I think we should leave this out

Reply:
But, if it won't break anything, being less strict here seems fine. I'm not sure how the `if len(s)` guard around the "some" return path in the original would be generalized, though (e.g. to floats/ints)